src/tests/Loader/classloader/regressions/GitHub_45600/repro.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+
+public abstract class A { }
+
+public class B : A { }
+
+public abstract class C { }
+
+public abstract class C<CTParam> : C where CTParam : A { }
+
+public class D : C<B> { }
+
+public abstract class E
+{
+    internal E() { }
+
+    internal abstract Type NamedObjectType { get; }
+}
+
+public class E<ETParam> : E
+    where ETParam : A
+{
+    private readonly F<C<ETParam>> components =
```

Comment:
Based on the fix I wonder if this repro can be further simplified to use as the test case for better readability?

Reply:
Well, yes, that's why I've pushed the case into a regressions area instead of a well named test case.